initial draft of branching config behaviour (#30548)
supabase/supabase: apps/docs/content/guides/deployment/branching.mdx

  }}
/>

Branching uses the `config.toml` file in your `supabase` directory to prepare and deploy the different services to each branch. These are:

> 1. Database - Using the `migrations` directory for schema changes and applying the contents of `seed.sql` for seed data

Reviewer: ```suggestion
1. Database - Using the `migrations` directory for schema changes and `[db.seed]` block to declare the `sql_paths` for seeding data. 
```

2. Auth - The `auth` section of the `config.toml` file is used to sync your project config with the Branch
3. Storage - Buckets defined in the  `storage` section of the `config.toml` are created upon deployment of a new Branch and are seeded with files within the directory defined in the `objects_path` property

> 4. Edge Functions - Functions within the `functions` directory are deployed

Reviewer: ```suggestion
4. Edge Functions - Functions are declared in `[functions.my-slug]` block with `entrypoint_path` specifying the `index.ts` or `index.js` file to your edge function.
```

3. Storage - Buckets defined in the  `storage` section of the `config.toml` are created upon deployment of a new Branch and are seeded with files within the directory defined in the `objects_path` property
4. Edge Functions - Functions are declared in `[functions.my-slug]` block with `entrypoint_path` specifying the location of `index.ts` or `index.js` file to your Edge Function.

<Tabs
  scrollable
  size="small"
  type="underlined"
  defaultActiveId="auth"
  queryGroup="platform"
>
<TabPanel id="api" label="API Config Example">
```toml
[api]
enabled = true
schemas = ["public","graphql_public","storage"]
```
</TabPanel>
<TabPanel id="auth" label="Auth Config Example">
```toml
[auth.external.github]
enabled = true
client_id = "env(GITHUB_CLIENT_ID)"
secret = "env(GITHUB_SECRET)"
redirect_uri = "" # Overrides the default auth redirectUrl.
```
</TabPanel>
<TabPanel id="storage" label="Storage Config Example">
```toml
[storage]
file_size_limit = "50MiB"
[storage.buckets.avatars]
public = false
allowed_mime_types = ["image/png", "image/jpeg"]
```
</TabPanel>
<TabPanel id="function" label="Edge Function Config Example">
```toml
[functions.hello-world]

> entrypoint_path="index.js"

Reviewer: ```suggestion
entrypoint = "functions/hello-world/index.js"
```

All paths are relative to config.toml file.
